pkg/maps/egressmap/egress.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2020 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package egressmap
+
+import (
+	"fmt"
+	"net"
+	"unsafe"
+
+	"github.com/cilium/cilium/pkg/bpf"
+	"github.com/cilium/cilium/pkg/types"
+)
+
+const (
+	MapName    = "cilium_egress_v4"
+	MaxEntries = 16384
+)
+
+// +k8s:deepcopy-gen=true
+// +k8s:deepcopy-gen:interfaces=github.com/cilium/cilium/pkg/bpf.MapKey
+type Key4 struct {
+	PrefixLen uint32
+	SourceIP  types.IPv4
+	DestCIDR  types.IPv4
+}
+
+func (k *Key4) GetKeyPtr() unsafe.Pointer { return unsafe.Pointer(k) }
+func (k *Key4) NewValue() bpf.MapValue    { return &EgressInfo4{} }
+func (k *Key4) String() string {
+	return fmt.Sprintf("%s %s/%d", k.SourceIP, k.DestCIDR, k.PrefixLen-getStaticPrefixBits())
+}
+
+func getStaticPrefixBits() uint32 {
```

Comment:
Perhaps I'm missing something, but why not just use a constant for that since this will always return 32 bits (alias `sizeof(ipv4) * 8`)?

Reply:
Actually I copied the same pattern from ipcache, which is pretty similar: 

https://github.com/cilium/cilium/blob/d9fa628eabe3568c2dee2405265e5ef2c2c7464d/pkg/maps/ipcache/ipcache.go#L60-L67

https://github.com/cilium/cilium/blob/d9fa628eabe3568c2dee2405265e5ef2c2c7464d/pkg/maps/ipcache/ipcache.go#L76-L81